### PR TITLE
[SPARK-44961][PYTHON][CONNECT][TESTS] Make PySpark (pyspark-connect module) tests passing without any dependency

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR proposes to fix the tests to properly run or skip when there aren't optional dependencies installed.

### Why are the changes needed?

Currently, it fails as below:

```
./python/run-tests --python-executables=python3 --modules=pyspark-connect
...
2c5289024a/python3__pyspark.sql.connect.window__nvbbzy7q.log)
Finished test(python3): pyspark.sql.connect.session (0s)
Traceback (most recent call last):
  File "/Users/hyukjin.kwon/workspace/forked/spark/python/pyspark/sql/pandas/utils.py", line 27, in require_minimum_pandas_version
    import pandas
ModuleNotFoundError: No module named 'pandas'
```

PySpark tests should pass without optional dependencies.

### Does this PR introduce _any_ user-facing change?

No, test-only.

### How was this patch tested?

Manually ran as described above.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

File: python/pyspark/sql/connect/streaming/query.py [hunk 1]
```diff
@@ -33,8 +36,6 @@
     StreamingQueryException as CapturedStreamingQueryException,
 )
 
-__all__ = ["StreamingQuery", "StreamingQueryManager"]
```
Comment (HyukjinKwon) [inline python/pyspark/sql/connect/streaming/query.py:36]: All `connect` is an internal module. So I we don't need this

File: python/pyspark/sql/connect/streaming/__init__.py [hunk 2]
```diff
@@ -14,9 +14,3 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-
-from pyspark.sql.connect.streaming.query import StreamingQuery  # noqa: F401
-from pyspark.sql.connect.streaming.readwriter import DataStreamReader  # noqa: F401
-from pyspark.sql.connect.streaming.readwriter import DataStreamWriter  # noqa: F401
-from pyspark.sql.connect.streaming.query import StreamingQueryManager  # noqa: F401
-from pyspark.errors import StreamingQueryException  # noqa: F401
```
Comment (HyukjinKwon) [inline python/pyspark/sql/connect/streaming/__init__.py:22]: All `connect` is an internal module. So I we don't need this

**Review by zhengruifeng — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM (Pending CIs). Thank you so much for this, @HyukjinKwon .

**Comment by dongjoon-hyun** [comment]
I verified manually. Thank you, @HyukjinKwon and @zhengruifeng .
Merged to master according to the affected version of JIRA.

**Pull request closed without merging**